cmake.deps/CMakeLists.txt

```diff
@@ -110,7 +110,14 @@ if(USE_BUNDLED_UNIBILIUM)
 endif()
 
 if(USE_BUNDLED_LIBUV)
-  include(BuildLibuv)
+  include(CheckSymbolExists)
+  check_symbol_exists(pthread_getname_np "pthread.h" HAVE_PTHREAD_GETNAME_NP)
+
+  if(HAVE_PTHREAD_GETNAME_NP)
+    include(BuildLibuv)
+  else()
+    set(USE_BUNDLED_LIBUV OFF)
+  endif()
```

Comment:
I would prefer this check to be at the top with the "# Options" comment, then it can be rewritten as the following:

```cmake
include(CheckSymbolExists)
check_symbol_exists(pthread_getname_np "pthread.h" HAVE_PTHREAD_GETNAME_NP)
option(USE_BUNDLED_LIBUV "Use the bundled libuv." ${USE_BUNDLED} AND ${HAVE_PTHREAD_GETNAME_NP})
```

That last line might be need to be rewritten as a basic if-def like it's done with USE_BUNDLED_GETTEXT, but that's still easier IMO.